SDWebImage/UIView+WebCache.h

```diff
@@ -50,6 +51,15 @@ typedef void(^SDSetImageBlock)(UIImage * _Nullable image, NSData * _Nullable ima
                           progress:(nullable SDWebImageDownloaderProgressBlock)progressBlock
                          completed:(nullable SDExternalCompletionBlock)completedBlock;
 
+- (void)sd_internalSetImageWithURL:(nullable NSURL *)url
```

Comment:
This args seems a little ugly. Can we detect that `[self isKindOfClass:[FLAnimatedImageView class]` to determine this setBlock should happed in global queue instead of main queue ?

Reply:
We can't even do that :( the part where we set the image on needs to be on the main queue, only the part creating the `FLAnimatedImage` needs to be on a background queue.
That's why I said I don't like it, it adds complexity I don't think we need.
Do you see any other way around?

Reply:
...A question, why all our public method happed on main queue :) . Even `-[SDWebImageManager loadImageWithURL:options:progress:completed:]`. A better solution is to add some custom code for FL in `UIView+WebCache.m`. Since this is currently the only way we can find that `[self class]`.(Or we need to attach a flag or context ,this is too more tricky)

Reply:
@dreampiggy I don't understand your question :)
What do you mean "all our public method happen on the main queue"?
And regarding the custom FL code, can you give a code example?

Reply:
@bpoplauschi I mean many check with that `dispatch_main_async_safe` is a little misunderstanding. (Previously I think this is happed in a global queue. But I find that nearly all the callback in our API such as SDWebImageManager, SDWebImageDownloader are dispatched on the main queue). Anyway, I try another approach without modifying header files.

Reply:
I think users expect completion blocks to be executed on the main queue, especially for UI components like `UIImageView` where you will add UI code. If we don't, we get a lot of complaints :) also, we can't change this now after it's been like this for so long.
Maybe it's only my opinion :)